### PR TITLE
fix(generator source): Give the generator source a default interval

### PR DESCRIPTION
**Pull request by lucperkins**
This may be somewhat controversial, in which case I'll retract it. But I feel like the default behavior of the `generator` source is probably not in line with users' expectations. As it stands, running the `generator` without specifying `interval` means that your terminal is *flooded* with a dizzying stream of events. Making the default behavior a steady trickle seems better to me. The 1.0 figure is purely arbitrary but seems intuitive enough.

I'm happy to retract this if others think that zero delay is indeed aligned with user expectations.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
👍 makes sense to me. I do think this is a better default behavior for users.

File: src/sinks/util/adaptive_concurrency/tests.rs [hunk 1]
```diff
@@ -103,7 +103,7 @@ struct TestParams {
     requests: usize,
 
     // The time interval between requests.
-    interval: Option<f64>,
```
Comment (jszwedko) [inline src/sinks/util/adaptive_concurrency/tests.rs:106]: I don't think you want to modify this. This is unrelated to the generator.

Comment (lucperkins) [reply src/sinks/util/adaptive_concurrency/tests.rs:106]: No, this is actually passed into a `GeneratorConfig` down below: https://github.com/vectordotdev/vector/blob/master/src/sinks/util/adaptive_concurrency/tests.rs#L405-L406

**Comment by spencergilbert** [comment]
Does this need to be noted as a breaking change?

**Comment by jszwedko** [comment]
> Does this need to be noted as a breaking change?

Yeah, good call, we should note it even if the impact is low. @lucperkins you can add it to https://github.com/vectordotdev/vector/blob/master/website/content/en/highlights/2021-10-05-0-17-upgrade-guide.md

File: src/sinks/util/adaptive_concurrency/tests.rs (hunk 1 above)
Comment (jszwedko) [reply src/sinks/util/adaptive_concurrency/tests.rs:106]: Aha! I stand corrected. You'll need to update the test fixtures in `tests/data/adaptive-concurrency/`. It is currently causing a failure since this change makes the `interval` field required:

https://github.com/vectordotdev/vector/pull/9390/checks?check_run_id=3759479954#step:9:141

I think you'll want to specify an interval of 0 to match the previous behavior when `interval` wasn't in the fixture file.

Alternatively, you could add a serde default of 0 for the `interval` here for the concurrency tests.

**Comment by lucperkins** [comment]
@spencergilbert @jszwedko Okay, I added a note in the upgrade guide.

File: src/sinks/util/adaptive_concurrency/tests.rs (hunk 1 above)
Comment (lucperkins) [reply src/sinks/util/adaptive_concurrency/tests.rs:106]: Good call. I went with a serde default here, as that seems to be the least disruptive.

**Review by jszwedko — APPROVED** [review]
Thanks for adding the upgrade note!

**Comment by lucperkins** [comment]
@jszwedko The test failure *seems* to be spurious, as it's only on one platform (Mac) and in an area that's both unrelated to these changes and a persistent source of 😿 (filesystem stuff). I'm going to merge and keep an eye out.
